### DISCUSSION TITLE
[Beginner] Download help

### DISCUSSION THREAD

**Discussion by NohaChoi** [sherlock-project/sherlock#2230] [Q&A] [answered]
Hello this is just another one of those complete beginner being confused posts. I am on windows and using vscode and trying to install/run the program(?). 

I've done the following so far.

downloaded git and python
git clone https://github.com/sherlock-project/sherlock.git
cd sherlock
sherlock testname1

For some reason the command isn't recognized, any ideas why?
![image](https://github.com/user-attachments/assets/b3565f51-3191-4824-a016-a109894d1e0e)

I have also tried the same process on cmd to the same results.
Thank you and will try to respond quickly.

**Comment by ppfeister** [comment]
Neither vscode nor git should be necessary (cloning is an older method, but no longer supported)

I'd point you to the Readme for current methods but  I don't think Windows is actually documented yet... we should add that...

On Windows, assuming you have Python installed...this probably works:

```python
python -m pip install sherlock-project
python -m sherlock_project user123
```

**Reply by NohaChoi** [reply]
Running the first command results in 
![image](https://github.com/user-attachments/assets/bf6fce2e-685f-4710-928a-c147f3581902)
lots of "Requirement satisfied"

Unfortunately this is the result I am getting from running the 2nd command.

![image](https://github.com/user-attachments/assets/a9db4d5b-2041-4a64-a5f0-73081a385abc)

**Reply by ppfeister** [reply, marked as answer]
Typo on my end (just edited my first reply to fix it)

While the package is `sherlock-project`, the importable module is `sherlock_project`. Hyphens normally translate to underscores.

Try `python -m sherlock_project ...` and let me know how it goes

On most platforms this isn't necessary -- Windows just likes to be special

**Reply by NohaChoi** [reply]
Thank you !!

**Reply by TauheedCodez** [reply]
Thanks !!

**Reply by sinatra-sell** [reply]
thanks!!

**Reply by evansmm000-sys** [reply]
Sherlock offers several installation methods, including `pipx`, `pip`, `apt`, `dnf`, `brew`, and Docker . The `pipx` method is officially supported and maintained by the Sherlock Project itself

**Comment by michaelploberger** [comment]
🌵🐝
